packages/DataTable/src/components/InfiniteLoader/InfiniteLoader.tsx

```diff
@@ -63,12 +65,36 @@ export function InfiniteLoaderImpl({
   isNextPageLoading = false,
   minimumBatchSize = 10,
   threshold = 15,
+  resetDimension,
 }: InfiniteLoaderPrivateProps & InfiniteLoaderPublicProps): JSX.Element {
   const infiniteLoaderRef = React.useRef(null);
   const listRef = React.useRef<VariableSizeList>(null);
-  const { getItemSize } = useItemSizeCalculator(data, getRowHeight);
+  const isLoadingMoreItemsRef = React.useRef(false);
+  const prevData = React.useRef<TableDataItemType[]>(data);
+  const { getItemSize, clearRowHeights } = useItemSizeCalculator(data, getRowHeight);
+
+  React.useLayoutEffect(() => {
+    if (!isLoadingMoreItemsRef.current && listRef.current) {
+      const changedIndexes: number[] = [];
+      prevData.current.forEach((row: TableDataItemType, index: number) => {
+        if (!isEqual(row, data[index])) {
+          changedIndexes.push(index);
+        }
+      });
+      if (changedIndexes.length > 0) {
+        clearRowHeights(changedIndexes);
+        listRef.current.resetAfterIndex(changedIndexes[0]);
+        setTimeout(resetDimension, 0);
```

Comment:
Finding i've had to use setTimeout here to ensure the resize gets applied after the react table has finished resetting, I don't see a way to avoid it